thriftbp/client_middlewares.go

```diff
@@ -361,11 +359,10 @@ func PrometheusClientMiddleware(remoteServerSlug string) thrift.ClientMiddleware
 				clientActiveRequests.With(activeRequestLabels).Inc()
 
 				defer func() {
-					var exceptionTypeLabel, baseplateStatusCode, baseplateStatus string
+					var baseplateStatusCode, baseplateStatus string
+					exceptionTypeLabel := stringifyErrorType(err)
 					success := strconv.FormatBool(err == nil)
 					if err != nil {
-						exceptionTypeLabel = strings.TrimPrefix(fmt.Sprintf("%T", err), "*")
-
 						var bpErr baseplateError
 						if errors.As(err, &bpErr) {
```

Comment:
ah oops. This will never match because `err` is a `thrift.TException` not a `baseplateError`

Reply:
sorry I'm discovering this after you merged

Reply:
no an concrete error type can implement 2 interfaces.

Reply:
well I just ran it through the debugger and it doesn't enter the `if errors.As(err, &bpErr) {` block

Reply:
I just created a repro in auth-go which uses `thrift_version=9bee877` and crafted a request so that the response is a `Baseplate.Error`.

adding a break point here, I can see that `err` is `github.com/apache/thrift/lib/go/thrift.TException(*github.snooguts.net/reddit/reddit-service-authentication-go/thrift/reddit/baseplate.Error) *{Code: *401, Message: *"Unauthorized", Details: map[string]string ["reason": "could not find user Identity. NotFoundError({})", ]}` and then stepping through execution I can see it pass over the `if errors.As(err, &bpErr) {...}` block.

Reply:
[T](https://github.com/reddit/baseplate.go/pull/485) is what I mean